src/panels/lovelace/cards/hui-shopping-list-card.ts

```diff
@@ -131,24 +170,46 @@ class HuiShoppingListCard extends hassLocalizeLitMixin(LitElement)
           position: relative;
           top: 1px;
         }
+        .label {
+          color: var(--primary-color);
+          margin-left: 17px;
+          margin-bottom: 11px;
+          margin-top: 11px;
+        }
+        .divider {
+          height: 1px;
+          background-color: var(--divider-color);
+          margin: 10px;
+        }
       </style>
     `;
   }
 
   private async _fetchData(): Promise<void> {
-    if (this._hass) {
-      this._items = await fetchItems(this._hass);
+    if (this.hass) {
+      const checkedItems: ShoppingListItem[] = [];
+      const uncheckedItems: ShoppingListItem[] = [];
+      const items = await fetchItems(this.hass);
+      for (const key in items) {
+        if (items[key].complete) {
+          checkedItems.push(items[key]);
+        } else {
+          uncheckedItems.push(items[key]);
+        }
+      }
+      this._checkedItems = checkedItems;
+      this._uncheckedItems = uncheckedItems;
     }
   }
 
   private _completeItem(ev): void {
-    completeItem(this._hass!, ev.target.itemId, ev.target.checked).catch(() =>
+    completeItem(this.hass!, ev.target.itemId, ev.target.checked).catch(() =>
       this._fetchData()
     );
   }
 
   private _saveEdit(ev): void {
```

Comment:
Param Type

Reply:
ech. I don't know how to handle this. It's a MouseEvent...but but the target is then a paper-checkbox type with an additional field `itemId` that I added...

Reply:
actually _saveEdit would be a KeyboardEvent I guess...but only applies when you either click out or hit enter...I don't know what to do...